docs/DeveloperGuide.md

```diff
@@ -175,6 +175,28 @@ The activity diagrams below detail the behaviour of PayMeLah when a user inputs
 <img src="images/AddDebtActivityDiagram.png" width="450" />
 <img src="images/AddDebtActivityDiagramRake.png" width="450" />
 
+### Split debt feature: `splitdebt`
+
+#### Implementation
+
+This feature is facilitated by `SplitDebtCommand` and `SplitDebtCommandParser` in the `Logic` component, and work as per [described above](#logic-component).
+
+When given a valid user input, the `SplitDebtCommandParser` will create a new `Debt` object to add to the `DebtList` of the multiple specified `Person`s.
+The following implementation is highly similar to the above implementation of [add debt](#add-debt-feature-adddebt), however, it differs in that '0' is a valid position in representing a `Person` to the `SplitDebtCommandParser`, and
+the value of the `Money` of the `Debt` added to each `Person` is the amount the command specified divided by the number of specified `Person` objects.
+However, '0' is not a valid `Index` internally and is only a placeholder to indicate that the user is included in the number of specified `Person` objects. Thus, no `Debt` is actually added to the `Person` whose position in the
+displayed list is '0'. It should also be noted that there is also never such a `Person`. The non-zero `Index` objects of the `Person` objects splitting the `Debt` are contained in a set `DeleteDebtCommandParser` creates.
+The `SplitDebtCommandParser` will construct a `SplitDebtCommand` object with this set of `Index` objects and the `Debt` object created.
+
+Consider a scenario where the user wishes to split a $30 food debt amongst multiple people. To speed up adding similar `Debt` objects to the `DebtList` of more than 1 `Person`, the `SplitDebtCommand` uses the set of `Index` objects. For each `Person` that an `Index` corresponds to, a new `Debt` object will be added to the `DebtList` of the specified `Person`.
+Some commands such as marking debts as paid, or possible future extensions such as editing a debt, may require modifying a `Debt` object in the `DebtList` of a `Person`. To ensure that modifying this `Debt` for 1 `Person` does not also erroneously modify the `Debt` of another `Person`, during execution of `splitdebt`, each `Debt` object is only added to one `DebtList`, and an `equal` instance of `Debt` is created and added for each other `DebtList`.
+
+To enable the user to retroactively add a `Debt` that is backdated, the `SplitDebtCommandParser` can take in optional `<date>` and `<time>` parameters. By making these parameters optional, a default behaviour can be implemented such that when neither parameter is specified, a `Debt` object with the current date and time is created. This will improve the efficiency at which users can input new `Debt` objects for the (expected) most common scenario where they add the `Debt` into PayMeLah on the actual day the debt occurred.
```

Comment:
this one is also copied, can shorten or remove since it does not differ from adddebt.